### PR TITLE
Rephrase the error message for missing value

### PR DESCRIPTION
**Pull request by melix**
### Context

Use the new mechanism for throwing an error and document the problem.

### CODE REVIEW

**Comment by melix** [comment]
@bot-gradle test QFL

File: subprojects/core/src/test/groovy/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTaskFactoryTest.groovy [hunk 1]
```diff
@@ -49,61 +52,11 @@ import spock.lang.Unroll
...
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TaskWithOverloadedIncrementalAndInputChangesActions
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TaskWithOverloadedInputChangesActions
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TaskWithOverriddenIncrementalAction
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TaskWithOverriddenInputChangesAction
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TaskWithOverriddenMethod
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TaskWithProtectedMethod
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TaskWithSingleParamAction
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TaskWithStaticMethod
-import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.TestTask
+import static org.gradle.api.internal.project.taskfactory.AnnotationProcessingTasks.*
```
Comment (melix) [inline subprojects/core/src/test/groovy/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTaskFactoryTest.groovy:55]: Will fix after review, I don't want to trigger a full build just for this.

**Comment by blindpirate** [comment]
@bot-gradle test QFL

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**